tests/kernel/timer/timer_api/src/main.c

```diff
@@ -267,6 +267,46 @@ void test_timer_period_0(void)
 	k_timer_stop(&period0_timer);
 }
 
+/**
+ * @brief Test Timer with K_FOREVER period value
+ *
+ * Validates initial timer duration, keeping timer period to K_FOREVER.
+ * Basically, acting as one-shot timer.
+ * It initializes the timer with k_timer_init(), then starts the timer
+ * using k_timer_start() with specific initial duration and period as
+ * zero. Stops the timer using k_timer_stop() and checks for proper
+ * completion.
+ *
+ * @ingroup kernel_timer_tests
+ *
+ * @see k_timer_init(), k_timer_start(), k_timer_stop(), k_uptime_get(),
+ * k_busy_wait()
+ */
+void test_timer_period_k_forever(void)
+{
+	init_timer_data();
+	/** TESTPOINT: set period 0 */
+	k_timer_start(
+		&period0_timer,
+		K_TICKS(k_ms_to_ticks_floor32(DURATION) -
+			BUSY_SLEW_THRESHOLD_TICKS(DURATION * USEC_PER_MSEC)),
+		K_FOREVER);
+	tdata.timestamp = k_uptime_get();
+
+	// Need to wait at least 2 durations to ensure one-shot behavior.
+	busy_wait_ms(DURATION + 1);
```

Comment:
`busy_wait_ms(2 * DURATION + 1)` would be more clear, IMO.

Also in the next commit.

Reply:
For some reason I thought that doing this still masked the issue but I just reran with this change and it works as expected, my mistake. Thanks!